geode-core/src/test/java/org/apache/geode/internal/tcp/ConnectionJUnitTest.java

```diff
@@ -82,4 +87,21 @@ public void testSuspicionRaised() throws Exception {
     conn.run();
     verify(membership).suspectMember(isNull(InternalDistributedMember.class), any(String.class));
   }
+
+  @Test
+  public void connectTimeoutIsShortWhenAlerting() throws UnknownHostException {
+    ConnectionTable table = mock(ConnectionTable.class);
+    TCPConduit conduit = mock(TCPConduit.class);
+    when(table.getConduit()).thenReturn(conduit);
+    when(conduit.getSocketId())
+        .thenReturn(new InetSocketAddress(SocketCreator.getLocalHost(), 12345));
+    DistributionConfig config = mock(DistributionConfig.class);
+    when(config.getMemberTimeout()).thenReturn(100);
+    Connection connection = new Connection(table, mock(Socket.class));
```

Comment:
I added this mock to make this test pass with `Connection.java` as it existed pre-PR:

```java
    DistributionManager distMgr = mock(DistributionManager.class);
    when(distMgr.getConfig())
        .thenReturn(config);
    when(conduit.getDM())
        .thenReturn(distMgr);
```

Reply:
sorry but I can't reproduce your work.  If I revert connection.java and add your mocks to the test I need to make the getP2PConnectTimeout non-private.  Then I run the test and it fails with the exception I showed above.

``` @Test
  public void connectTimeoutIsShortWhenAlerting() throws UnknownHostException {
    ConnectionTable table = mock(ConnectionTable.class);
    TCPConduit conduit = mock(TCPConduit.class);
    when(table.getConduit()).thenReturn(conduit);
    when(conduit.getSocketId())
        .thenReturn(new InetSocketAddress(SocketCreator.getLocalHost(), 12345));
    DistributionConfig config = mock(DistributionConfig.class);
    when(config.getMemberTimeout()).thenReturn(100);
    DistributionManager distMgr = mock(DistributionManager.class);
    when(distMgr.getConfig())
        .thenReturn(config);
    when(conduit.getDM())
        .thenReturn(distMgr);
    Connection connection = new Connection(table, mock(Socket.class));
    int normalTimeout = connection.getP2PConnectTimeout();
    AlertingAction.execute(() -> {
      assertThat(normalTimeout).isEqualTo(6 * connection.getP2PConnectTimeout());
    });

  }```

Reply:
It does look like I have my assertion values backward though

Reply:
It looks like the assertions should be more like this:
```    int normalTimeout = connection.getP2PConnectTimeout();
    assertThat(normalTimeout).isEqualTo(600);
    AlertingAction.execute(() -> {
      assertThat(connection.getP2PConnectTimeout()).isEqualTo(3600);
    });
```

Reply:
the assertions need some work - I'll check into it Bill

Reply:
The assertions should be more like this:
```    Connection connection = new Connection(table, mock(Socket.class));
    int normalTimeout = connection.getP2PConnectTimeout(config);
    assertThat(normalTimeout).isEqualTo(600);
    AlertingAction.execute(() -> {
      assertThat(connection.getP2PConnectTimeout(config)).isEqualTo(100);
    });
```